src/components/Search/SearchBox.js

```diff
@@ -8,6 +8,17 @@ export default class SearchBox extends Component {
         super(props);
         this._handleKeyPress = this._handleKeyPress.bind(this);
     }
+    componentDidMount() {
+        if (location.hash.indexOf('/search/') !== -1) {
+            const searchQuery = location.hash.substr(9);
+            if (searchQuery.length > 0) {
+                this.props.titleClick();
+                this.props.valChange(searchQuery);
+                document.getElementsByClassName('searchInput')[0].value = searchQuery;

```

Comment:
Why you need to to this? If you're using the `valChange` prop it should be done by default. You already have the value of this input mapped in the state, so if you update the state it will already update the input for you, not needed for this.

Reply:
Yes, you are right. This line is not needed.